### PR TITLE
Initial version with first utils

### PR DESCRIPTION
**Pull request by Jongy**
This includes extracting OOM events & exception signals (SIGSEGV, traps, ...) from kernel messages, which we plan to use in gProfiler among other components.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+
+version = re.search(r'__version__\s*=\s*"(.*?)"', Path("utils/__init__.py").read_text())
+assert version is not None, "could not parse version!"
+
+with open("README.md", "r", encoding="utf-8") as fh:
+    long_description = fh.read()
+
+setuptools.setup(
+    name="utils",
```
Comment (d3dave) [inline setup.py:24]: That's gonna conflict with https://pypi.org/project/utils/

**Review by d3dave — CHANGES_REQUESTED** [review]

File: setup.py (hunk 1 above)
Comment (Jongy) [reply setup.py:24]: Wasn't planning to, at least initially. We can install directly from git.

Comment (d3dave) [reply setup.py:24]: pip is gonna complain if someone installs that package.

Comment (Jongy) [reply setup.py:24]: Our components run in their own container / own Python environment. So I doubt it'll affect us. We can name it `granulate-utils`.

File: lint.sh [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+if [ -f venv/bin/activate ]; then
+  source venv/bin/activate
+fi
+
+check_arg=""
+if [[ "$1" = "--ci" ]]; then
+    check_arg="--check"
+fi
+
+isort --settings-file .isort.cfg .
```
Comment (d3dave) [inline lint.sh:13]: Which version of `isort` are you using? New version uses `--settings-path` and requires `-rc`

File: .flake8 [hunk 3]
```diff
@@ -0,0 +1,5 @@
+[flake8]
+max-line-length = 120
+# options for compatibility with black - see https://black.readthedocs.io/en/stable/compatible_configs.html#flake8
+extend-ignore = E203, W503
+exclude = .git,__pycache__,build,gprofiler.egg-info,venv
```
Comment (d3dave) [inline .flake8:5]:
```suggestion
exclude = .git,__pycache__,build,venv
```

**Review by d3dave — CHANGES_REQUESTED** [review]

File: .flake8 (hunk 3 above)
Comment (Jongy) [reply .flake8:5]: Hmm already had it but I didn't push
```
commit ea5b5fff91a365e30355674d5ed9edfd75837214 (HEAD -> first-utils)
Author: Yonatan Goldschmidt <yonatan.goldschmidt@granulate.io>
Date:   Wed Nov 17 02:13:19 2021 +0200

    Remove unrelated dirs

diff --git a/.flake8 b/.flake8
index 2b3aecf..4dacb92 100644
--- a/.flake8
+++ b/.flake8
@@ -2,4 +2,4 @@
 max-line-length = 120
 # options for compatibility with black - see https://black.readthedocs.io/en/stable/compatible_configs.html#flake8
 extend-ignore = E203, W503
-exclude = .git,__pycache__,build,gprofiler.egg-info,venv
+exclude = .git,__pycache__,venv
```

File: lint.sh (hunk 2 above)
Comment (Jongy) [reply lint.sh:13]: VERSION 5.8.0.

Will update both

Comment (Jongy) [reply lint.sh:13]:
```
/home/jong/.local/lib/python3.8/site-packages/isort/main.py:1206: UserWarning: W0501: The following deprecated CLI flags were used and ignored: -rc!
  warn(
/home/jong/.local/lib/python3.8/site-packages/isort/main.py:1210: UserWarning: W0500: Please see the 5.0.0 Upgrade guide: https://pycqa.github.io/isort/docs/upgrade_guides/5.0.0/
  warn(
```

not adding `-rc`.

Comment (d3dave) [reply lint.sh]: `--recursive` otherwise it doesn't recurse

Comment (Jongy) [reply lint.sh]: It's not a flag to 5.8.0
```
$ isort --help | grep -2 -- recursive
                        against.
  --dont-follow-links   Tells isort not to follow symlinks that are
                        encountered when running recursively.
  --filename FILENAME   Provide the filename associated with a stream.
```

File: setup.py (hunk 1 above)
Comment (Jongy) [reply setup.py:24]: Let's make this change if this ever bothers

**Comment by Jongy** [comment]
Rebased

File: setup.py (hunk 1 above)
Comment (Jongy) [reply setup.py]: Changed in #6

**Review by d3dave — APPROVED** [review]

**Pull request merged**